lib/charms/vault_k8s/v0/vault_autounseal.py

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the vault-autounseal relation.
+
+This library contains the Requires and Provides classes for handling the
+vault-autounseal interface.
+
+The provider side of the interface is responsible for enabling the vault
+transit engine and creating the necessary keys and policies for an external
+vault to be able to autounseal itself.
+
+The requirer side of the interface is responsible for retrieving the necessary
+details to autounseal the vault instance, and configuring the vault instance to
+use them.
+
+## Getting Started
+
+From a charm directory, fetch the library using `charmcraft`:
+
+```shell
+charmcraft fetch-lib charms.vault_k8s.v0.vault_autounseal
+```
+
+### Provider charm
+
+The provider charm is the charm that provides a Vault instance that can be
+used to autounseal other Vault instances via the Vault transit backend.
+
+### Requirer charm
+
+The requirer charm is the charm that wishes to autounseal a Vault instance via
+the Vault transit backend.
+
+### Integration
+
+You can integrate both charms by running:
+
+```bash
+juju integrate <vault a>:vault-autounseal-provides <vault b>:vault-autounseal-requires
```

Comment:
Here we document the interface name which is dependent on how we name those in metadata.yaml. If we want to force those to be this value, we may want to add some content in the docs here that relates to metadata.yaml.

Reply:
That's a good point. I it doesn't really matter what they call it, I suppose. If the intention is unclear, do you have any suggestions for a better way to document this? Should I just remove the interface name entirely like we do with other libs?

Reply:
We can add a section in the requirer side of the lib with content similar to this: 

```
Add the following to `metadata.yaml`:
    requires:
      vault-autounseal-requires:
        interface: vault-autounseal
```

And same thing for the provider.

Reply:
Is this what you were thinking?

https://github.com/canonical/vault-k8s-operator/pull/346/commits/c4f31eaaf52fc32123a4ceeb2f377006529a6a71

Reply:
Yes, I just added a suggestion to fix the formatting + a typo